acs-edge/docs/edge-split.md

```diff
@@ -0,0 +1,248 @@
+are potentially long strings containing arbitrary characters, the Edge
+Agent assigns a data topic name to each address it is currently using.
+These MUST be short strings of numbers, letters and underscores.
+
+The Edge Agent MUST manage data topic names in such a way as to avoid
+problems with synchronisation, and MUST NOT assume a driver will react
+instantly to a new address configuration.
+
+### Driver configuration
+
+Configuration specific to a particular driver, containing connection
+information and credentials and so on. This is in JSON form and is
+passed to the Edge Agent as part of the Edge Agent configuration.
+
+### Edge Agent configuration
+
+The Edge Agent configuration is a JSON document managed by the Factory+
+infrastructure and retrieved by the Edge Agent. It contains information
+about configured drivers and instructions for mapping driver data
+packets to Sparkplug metrics.
+
+## Connection and authentication
+
+The Edge Agent provides an MQTT broker interface for drivers to
+communicate with. Currently (due to likely implementation restrictions)
+this will be an MQTT 3.1.1 broker supporting username/password
+authentication only. Where a driver is run outside of the edge cluster
+the Edge Agent broker port will need to be made available externally.
+
+The driver is an MQTT client. The driver MUST authenticate to the Edge
+Agent broker using its connection name as both username and client-id,
+and a password. This information must be supplied to the driver as part
+of its deployment.
+
+Drivers which run as standalone applications SHOULD accept the following
+environment variables:
+
+Name|Meaning
+---|---
+`EDGE_MQTT`|URL of the Edge Agent MQTT broker
+`EDGE_USERNAME`|Driver connection name
+`EDGE_PASSWORD`|MQTT password
+
+Drivers MUST accept and support `mqtt://` URLs, including understanding
+that the port defaults to 1883, and MAY accept `mqtts://`, `ws://` and
+`wss://` URLs.
+
+## MQTT data packets
+
+Under normal circumstances MQTT QoS 0 should be adequate and avoids
+additional overhead. However, to allow for situations where it is not,
+drivers SHOULD obey these conventions:
+
+* The driver SHOULD set up its subscriptions and wait for the SUBACKs
+  before publishing anything.
+
+* The driver SHOULD subscribe with QoS 2 to allow the Edge Agent the
+  choice of what QoS to use.
+
+* The Edge Agent MAY downgrade the subscription in the SUBACK, and MAY
+  publish with a lower QoS.
+
+* The driver SHOULD observe the QoS granted for the `conf` topic and use
+  this QoS for all its PUBLISH packets. The driver SHOULD also observe
+  the QoS on all packets received on that topic and update its current
+  PUBLISH QoS.
+
+Drivers MAY instead choose to publish and subscribe entirely at QoS 0.
+
+Packets MUST NOT be published with the RETAIN flag set. The broker MUST
+NOT allow retained messages.
+
+## MQTT topics
+
+All topics are under the namespace `fpEdge1`. Each driver then
```

Comment:
@amrc-benmorrow what is the significance of the `1` at the end here?

Reply:
It's a version number, to allow for a potential incompatible `fpEdge2` later.

Reply:
Ah, I see. Any objections to `V1` instead?